x/evm/keeper/fees_test.go

```diff
@@ -444,99 +473,133 @@ func (suite *KeeperTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
-				ChainID:   zeroInt.BigInt(),
-				Nonce:     1,
-				To:        &suite.address,
-				Amount:    amount,
-				GasLimit:  tc.gasLimit,
-				GasPrice:  gasPrice,
-				GasFeeCap: gasFeeCap,
-				GasTipCap: gasTipCap,
-				Accesses:  tc.accessList,
-			}
-			tx := evmtypes.NewTx(ethTxParams)
-			tx.From = tc.from
+				err = vmdb.Commit()
+				suite.Require().NoError(err, "Unexpected error while committing to vmdb: %d", err)
 
-			txData, _ := evmtypes.UnpackTxData(tx.Data)
+				ethTxParams := &evmtypes.EvmTxArgs{
+					ChainID:   zeroInt.BigInt(),
+					Nonce:     1,
+					To:        &suite.address,
+					Amount:    amount,
+					GasLimit:  tc.gasLimit,
+					GasPrice:  gasPrice,
+					GasFeeCap: gasFeeCap,
+					GasTipCap: gasTipCap,
+					Accesses:  tc.accessList,
+				}
+				tx := evmtypes.NewTx(ethTxParams)
+				tx.From = tc.from
+
+				txData, _ := evmtypes.UnpackTxData(tx.Data)
 
-			evmParams := suite.app.EvmKeeper.GetParams(suite.ctx)
-			baseFee := suite.app.EvmKeeper.GetBaseFee(suite.ctx, evmParams)
-			priority := evmtypes.GetTxPriority(txData, baseFee)
+				evmParams := suite.app.EvmKeeper.GetParams(suite.ctx)
+				baseFee := suite.app.EvmKeeper.GetBaseFee(suite.ctx, evmParams)
+				priority := evmtypes.GetTxPriority(txData, baseFee)
 
-			fees, err := keeper.VerifyFee(txData, evmtypes.DefaultEVMDenom, baseFee, false, false, suite.ctx.IsCheckTx())
-			if tc.expectPassVerify {
-				suite.Require().NoError(err, "valid test %d failed - '%s'", i, tc.name)
-				if tc.enableFeemarket {
-					baseFeeRes, err := suite.app.EvmKeeper.BaseFee(suite.ctx, &evmtypes.QueryBaseFeeRequest{})
-					suite.Require().NoError(err)
-					suite.Require().Equal(
-						fees,
-						sdk.NewCoins(
-							sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewIntFromBigInt(txData.EffectiveFee(baseFeeRes.BaseFee.BigInt()))),
-						),
-						"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
-					)
-					suite.Require().Equal(int64(0), priority)
+				fees, err := keeper.VerifyFee(txData, evmtypes.DefaultEVMDenom, baseFee, false, false, suite.ctx.IsCheckTx())
+				if tc.expectPassVerify {
+					suite.Require().NoError(err, "valid test %d failed - '%s'", i, tc.name)
+					if tc.enableFeemarket {
+						baseFeeRes, err := suite.app.EvmKeeper.BaseFee(suite.ctx, &evmtypes.QueryBaseFeeRequest{})
+						suite.Require().NoError(err)
+						suite.Require().Equal(
+							fees,
+							sdk.NewCoins(
+								sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewIntFromBigInt(txData.EffectiveFee(baseFeeRes.BaseFee.BigInt()))),
+							),
+							"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
+						)
+						if setup.decimals == evmtypes.Denom6Dec {
+							// priority will be > 0 due to scaling up the gasTip
+							// and feeCap values and keeping the same DefaultPriorityReduction. Considering priority has no effect ATM
+							suite.Require().Greater(priority, int64(0))
+						} else {
+							suite.Require().Equal(int64(0), priority)
+						}
```

Comment:
Does this means that the 6 decimals could cause wrong behaviour for the priority is enabled?

Reply:
the calculation remains the same. The priority will depend on the tipPrice / DefaultPriorityReduction.
In this tests cases we leave the DefaultPriorityReduction same as with 18 dec, so when scaling all from 6 to 18, the tipPrice > DefaultPriorityReduction. That's why we get a priority > 0